runner/logs.go

```diff
@@ -332,24 +329,16 @@ func postprocessAndSendLogs(ctx context.Context, server *state.Server, globalCol
 
 	for idx := range transientLogState.LogFiles {
 		logFile := &transientLogState.LogFiles[idx]
+		logFile.ByteSize = int64(logFile.LogLines[len(logFile.LogLines)-1].ByteEnd)
```

Comment:
Since we no longer have the original string at this point in the code, looking at the last log line's `ByteEnd` is the only way to know the file size. Maybe this should be moved to an earlier processing stage?

Reply:
Oh and in terms of this code's soundness, I vaguely remember that on production we do receive log files that don't contain any log lines, so this would panic. Should we really be submitting empty snapshots?

Reply:
Looks like 11% of all log files are empty:
```sql
select count(*), byte_size = 0 as empty from log_files_7d group by 2 order by 1;
  count   | empty 
----------+-------
  6079570 | t
 55500294 | f

Reply:
In https://github.com/pganalyze/collector/pull/607/commits/7c6433c56ed260589017bc28ec02c351e1cb737e I've fixed the potential panic, and also made it so we don't submit empty log snapshots.